lib/libc/Kconfig

```diff
@@ -43,7 +43,8 @@ config NEWLIB_LIBC_SUPPORTED
 config PICOLIBC_SUPPORTED
 	bool
 	depends on !NATIVE_APPLICATION
-	depends on !REQUIRES_FULL_LIBCPP || ("$(TOOLCHAIN_HAS_PICOLIBC)" = "y")
+	depends on !REQUIRES_FULL_LIBCPP
```

Comment:
This looks wrong -- the logic here is that you must use the toolchain picolibc when using the C++ library, but it looks like you've made picolibc completely incompatible with the C++ library.

Reply:
probably fixed, could you give this another spin?